script.lua

```diff
@@ -7733,16 +7733,93 @@ function onObjectDestroy(obj)
         end
     end
 end
-function getCardImageURL(card)
-    local cardData = card.getData()
-    if cardData.CustomDeck then
-        for _, data in pairs(cardData.CustomDeck) do
-            if data.NumWidth == 1 and data.NumHeight == 1 then
-                return data.FaceURL
+function getReminderLocation(params)
+    local obj = params.obj
+    if obj == nil then
+        return nil
+    end
+
+    -- Get it from the script state if it's there
+    local state = {}
+    if obj.script_state ~= "" then
+        state = JSON.decode(obj.script_state)
+    end
+    if state.reminder ~= nil then
+        return state.reminder
+    end
+
+    -- Otherwise, use a sensible default
+    if obj.type == "Card" then
+        return {
+            field = "FaceURL",
+            x = -0.17,
+            y = -0.45,
+            width = 1.85,
+            height = 2.59,
+        }
+    elseif obj.hasTag("Spirit") then
+        return {
+            field = "ImageSecondaryURL",
+            x = 0.63,
+            y = -0.23,
+            width = 2.40,
+            height = 1.60,
+        }
+    else
+        return nil
+    end
+end
+function getReminderImageAttributes(params)
+    local objSize = 200 -- The height/width of UI panel to be the same size as the object
+
+    local obj = params.obj
+    if obj == nil then
+        return {image = ""}
+    end
+
+    local location = params.location
+    if location == nil then
+        location = getReminderLocation({obj = obj})
+    end
+
+    local imageURL = ""
+    local data = obj.getData()
+    if obj.type == "Card" then
+        if data.CustomDeck then
```

Comment:
while i'm not sure this is applicable in current mod, but there does seem to be a gap in logic for cards that don't have a custom deck

Reply:
As far as I can tell, that's just the cards that are built in to TTS: the standard deck of 52 playing cards (plus joker) and the CardBots cards. So yeah, you can get an error by grabbing a playing card, giving it the `Minor`, `Major` or `Unique` tag, and then using the reminder hotkey on it. I can fix that, but I doubt it will come up.

Reply:
Upon closer inspection, the error isn't even thrown because of this. It's thrown because playing cards, seemingly uniquely among all objects you can spawn in TTS, have `nil` instead of the empty string as their `script_state`. I have no idea what's going on with that, but I guess I can account for it.

Reply:
Alright, fixing that means that generating reminder tokens for playing cards doesn't error. Their images are just blank white ('cause the `image` attribute in their UI `Image` tag isn't set), but I think that's the best we can possibly get, 'cause I don't think there's anywhere we can actually access the images that are used on the playing cards.

But the chance that someone needs a reminder token for the ace of spades while playing SI is slim, and at least it won't error now.

Reply:
yeah we probably don't need a deck of cards to have a proper image mask